src/components/LoaderButton.js

```diff
@@ -1,20 +1,40 @@
-import Button from "react-bootstrap/lib/Button";
+import { Button } from "@mui/material";
+import { makeStyles } from "@mui/styles";
 import Glyphicon from "react-bootstrap/lib/Glyphicon";
 
-export default ({
-  isLoading,
+const useStyles = makeStyles((theme) => ({
+  root: {
+    "& .MuiButtonBase-root": {
+      marginTop: theme.spacing(1),
+      fontSize: 16,
+    },
+  },
+}));
+
+export default function LoaderButton({
   text,
   loadingText,
+  isLoading,
+  disabled,
   className = "",
-  disabled = false,
-  ...props
-}) => (
-  <Button
-    className={`LoaderButton ${className}`}
-    disabled={disabled || isLoading}
-    {...props}
-  >
-    {isLoading && <Glyphicon glyph="refresh" className="spinning" />}
-    {!isLoading ? text : loadingText}
-  </Button>
-);
+  type = "button",
+  color = "primary",
+  variant = "contained",
+}) {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.root}>
+      <Button
```

Comment:
The error you're getting from the compiler on this component is coming from the "type", "color", and "variant" properties. In Material UI they're restricted to specific values, while in this code, even though the value you assign to each is one of the available values, TypeScript is seeing them as variables of type string that could theoretically be any value. 

The easiest way around this is to convert the file to TypeScript and add a few lines at the start to define those three types so that the compiler recognizes them as 'safe' options.

Reply:
I did this and imported the Theme type to avoid errors when using makeStyles. What I submitted was a working version without any types or TypeScript.

Reply:
Sorry, maybe I misunderstood! I'll look into it some more. Having contributed to the Material UI library on a similar issue in the past, I assumed (maybe too hastily) that it's an issue with the library that requires a workaround.